### ISSUE TITLE
Fix false positives with :not() and disallowInList: true in no-duplicate-selectors

### ISSUE THREAD

**Issue by EvgenyOrekhov** [stylelint/stylelint#4340]
> Clearly describe the bug

There are false positives in `no-duplicate-selectors` with `disallowInList: true` when using `:not()`.

> Which rule, if any, is the bug related to?

`no-duplicate-selectors`

> What CSS is needed to reproduce the bug?

e.g.

```css
input:not([type="button"]) {
  border: 2px;
}

[type="button"] {
  border: 1px;
}
```

> What stylelint configuration is needed to reproduce the bug?

e.g.

```json
{
  "rules": {
    "no-duplicate-selectors": [true, {"disallowInList": true}]
  }
}
```

> Which version of stylelint are you using?

`11.1.1`

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

CLI with `stylelint 'src/**/*.css'`.

Also this bug reproduces here: https://stylelint.io/demo#N4Igxg9gJgpiBcICWA7ADgVwC7xRLAFANpYCeaMAvADogBG2WEKtAugJQAEw1KnndCACdYQ+JwBMaAB4BuXgF9evEuSq0GWJixCtuvfoJEwxnAIwz5KJShAAaEAGdSKLAENpCcI8f3wzADMkAHMvHj5OWiEMABsYR1pxcP5+WjwAWigMNBikMDcsGHTHGDiwJiEEhE4SaJg7fRAoJEc3GJiIAHcASRQAGRasRM4sOoVWA04bBRAFIA.

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

No.

> What did you expect to happen?

No warnings to be flagged.

> What actually happened (e.g. what warnings or errors did you get)?

The following warnings were flagged:

```text
5:1 error Unexpected duplicate selector "[type="button"]", first used at line 1 (no-duplicate-selectors)
```

**Comment by hudochenkov** [comment]
Thanks for the report and for using the template.

**Comment by EvgenyOrekhov** [comment]
Appears to be fixed since the demo example is not reproducible anymore.
